platform/source/minimal-printf/mbed_printf_implementation.c

```diff
@@ -37,39 +37,9 @@ static char mbed_stdio_out_prev = 0;
 #endif
 
 #if CONSOLE_OUTPUT == CONSOLE_OUTPUT_UART
-#if DEVICE_SERIAL
-/*
- Serial initialization and new line replacement is a direct copy from mbed_retarget.cpp
- If the static modifier were to be removed, this part of the code would not be necessary.
-*/
-#include "hal/serial_api.h"
-
-static serial_t stdio_uart = { 0 };
-
-/* module variable for keeping track of initialization */
-static bool not_initialized = true;
-
-static void init_serial()
-{
-    if (not_initialized) {
-        not_initialized = false;
-
-        serial_init(&stdio_uart, STDIO_UART_TX, STDIO_UART_RX);
-#if MBED_CONF_PLATFORM_STDIO_BAUD_RATE
-        serial_baud(&stdio_uart, MBED_CONF_PLATFORM_STDIO_BAUD_RATE);
-#endif
-    }
-}
-
-#define MBED_INITIALIZE_PRINT(x) { init_serial(); }
-#define MBED_PRINT_CHARACTER(x) { serial_putc(&stdio_uart, x); }
-
-#else
 
 #define MBED_INITIALIZE_PRINT(x)
-#define MBED_PRINT_CHARACTER(x)
-
-#endif // if DEVICE_SERIAL
+#define MBED_PRINT_CHARACTER(x) { fputc(x, stdout); }
```

Comment:
Having done this, the `CONSOLE_OUTPUT_UART` option no longer means that, and really the whole option should be removed (for now).

We already have the retargetting mechanisms to make fputc go to SingleWireOutput, so it isn't needed here.

In future there could be some system option to hardcode fputc globally to actually be a particular console type, cutting back the retargetting generally, but that option wouldn't be processed here.

Reply:
The mechanism in the retargetting code is via mbed_override_console which needs to be implemented by the application.
Althought it is simpler to have the option to output to SWO using a config option `minimal-printf-console-output` as was done by the original minimal-printf library, I can remove this code for the sake of simplicity.

Reply:
If you want "easy-to-select SWO",  it shouldn't be bound into the "minimal-printf" - the same easy-selection mechanism should be available with full retargetting.

You could have an option which does the override-console for you. In network interfaces we have such options done non-centrally as `esp8266.provide-default` etc, which activates that particular `WiFiInterface::get_default_instance`. SWO could have an option that makes it do that with `mbed_override_console`.

(Often I'd expect a target to be selecting SWO, if that's the way the target's wired, not the application itself).

The hypothetical minimal-retargetting would have to have a parallel hardcoded-mechanism to pick up the known hardcoded retargets. Not sure what the implementation would look like, but it could certainly follow the easy-selection option, so that option just uses the generic mechanism with full retargetting, or selects a hardcoded thing in minimal retargetting.